src/services/suggestionDiagnostics.ts

```diff
@@ -156,7 +156,29 @@ namespace ts {
     }
 
     function isPromiseHandler(node: Node): node is CallExpression {
-        return isCallExpression(node) && (hasPropertyAccessExpressionWithName(node, "then") || hasPropertyAccessExpressionWithName(node, "catch"));
+        return isCallExpression(node) && (
+            hasPropertyAccessExpressionWithName(node, "then") && hasZeroOrOneNonNullArguments(node) ||

```

Comment:
Unless passing all but one argument as `undefined` is a common pattern, I'd probably just count them and disallow anything other than 1.

Reply:
There are existing tests that assert the refactor works with `.then(undefined).then(...)`, which is absurd, but easy to keep supporting 🤷‍♂️